src/middlewares.js

```diff
@@ -424,6 +432,59 @@ export function promiseEnforceMasterKeyAccess(request) {
   return Promise.resolve();
 }
 
+export const addRateLimit = (route, id) => {
+  const config = Config.get(id);
+  const express = config.expressApp;
+  addLimitForRoute(express, route);
+};
+
+export const handleRateLimit = (api, rateLimit) => {
+  const options = Array.isArray(rateLimit) ? rateLimit : [rateLimit];
+  for (const route of options) {
+    addLimitForRoute(api, route);
+  }
+};
+
+const addLimitForRoute = (api, route) => {
+  api.use(
+    route.path || '*',
+    rateLimit({
+      windowMs: route.windowMs,
+      max: route.max,
+      message: route.message ?? 'Too many requests.',
```

Comment:
Default values should not be set in code but automatically set when parsing the parse server options. That is why the DEFAULT is defined in the definitions. You can see other examples in the file where the option validation happens.